core/ipc/sink/kafka/server/src/main/java/org/opennms/core/ipc/sink/kafka/server/KafkaMessageConsumerManager.java

```diff
@@ -181,8 +195,12 @@ public void run() {
                                 if (sinkMessage.getTotalChunks() > sinkMessage.getCurrentChunkNumber() + 1) {
                                     continue;
                                 }
-                                messageInBytes = largeMessageCache.getIfPresent(messageId).toByteArray();
+                                byteString = largeMessageCache.getIfPresent(messageId);
+                                if(byteString != null) {
```

Comment:
If it is `null`, then the contents of `messageInBytes` aren't valid. Looks like there should be some additional logging / handling for that case here.